server/graphql/v2/mutation/OrderMutations.js

```diff
@@ -184,100 +193,60 @@ const orderMutations = {
     },
     async resolve(_, args, req) {
       const decodedId = getDecodedId(args.order.id);
+      const haveDetailsChanged = !isUndefined(args.amount) && !isUndefined(args.tier);
```

Comment:
Ideally, yes. I didn't want to change this mutation too much to avoid breaking changes, but its usage is a bit complex at the moment (for example with parameters being exclusive)